skfp/fingerprints/klekota_roth.py

```diff
@@ -4990,3 +181,82 @@ def transform(
             Array with fingerprints.
         """
         return super().transform(X, copy)
+
+    def _dict_to_node(self, d: dict) -> _PatternNode:
+        """
+        Recursively convert a dict representation of a pattern tree
+        into a _PatternNode tree.
+        """
+        node = _PatternNode()
+        node.smarts = d.get("smarts")
+        node.pattern_mol = Chem.MolFromSmarts(node.smarts) if node.smarts else None
+        node.is_terminal = d.get("is_terminal", False)
+        node.feature_bit = d.get("feature_bit")
+        node.atom_requirements = defaultdict(int, d.get("atom_requirements", {}))
```

Comment:
If there are atom requirements, they should be specified explicitly, right? So why `defaultdict` with default value 0?